src/v/cluster/partition.h

```diff
@@ -57,11 +57,8 @@ class partition : public ss::enable_lw_shared_from_this<partition> {
       ss::lw_shared_ptr<const archival::configuration>,
       ss::sharded<features::feature_table>&,
       ss::sharded<archival::upload_housekeeping_service>&,
-      std::optional<cloud_storage_clients::bucket_name> read_replica_bucket
-      = std::nullopt,
-      std::optional<
-        std::reference_wrapper<ss::sharded<experimental::cloud_topics::app>>> dp
-      = std::nullopt);
+      std::optional<cloud_storage_clients::bucket_name> read_replica_bucket,
+      std::reference_wrapper<ss::sharded<experimental::cloud_topics::app>> dp);
```

Comment:
nit: could probably drop the reference wrapper i suppose it was only necessary so that std::optional would work.

Reply:
done in https://github.com/redpanda-data/redpanda/compare/0e98e07bea10675d38e4f745f0009c8253b8fbbc..62a0daabd8b3d50d1c607fdee7eff8a5058a52fc